openshift/sandboxed-containers-operator
# Apply peerpod related MachineConfigs

Peer pod enablement requires creating a new kata runtime class and providing the configuration.toml. In OpenShift we need to create MachineConfig objects to create these files.
The required MachineConfig yaml files are bundled within the controller-manager image and the manager process reads the yaml and creates the required MachineConfig objects. This gives us the flexibility to keep updating the yaml as needed without hardcoding the struct in the code

Fixes: #[Kata-2130](https://issues.redhat.com//browse/Kata-2130)


**- Description of the problem which is fixed/What is the use case**
PeerPod enablement requires creating configuration for a new runtimeclass and a new kata configuration.toml.
This PR enables creation of the required configs via the operator
**- What I did**
Added the required MachineConfig yamls to the controller-manager image and used the same in the code
to create the required configurations

**- How to verify it**
1. Create a KataConfig with `enablePeerPods: true`
2. Wait for KataConfig to be created successfully and `kata-oc` MCP to be in Updated state. 
3. Login the worker node and check for `/etc/crio/crio.conf.d/50-kata-remote-cc` and `/opt/kata/configuration-remote.toml`
4. These files are created by the MachineConfigs

**- Description for the changelog**

Add peerpod related MachineConfigs
